app/code/community/Ebizmarts/MailChimp/Model/Observer.php

```diff
@@ -126,65 +225,128 @@ public function customerSaveBefore(Varien_Event_Observer $observer)
-        Mage::getModel('mailchimp/api_subscribers')->update($customer->getEmail(), $storeId);
         return $observer;
     }
 
     public function customerAddressSaveBefore(Varien_Event_Observer $observer)
     {
         $customerId = $observer->getEvent()->getCustomerAddress()->getCustomerId();
-        $customer = Mage::getModel('customer/customer')->load($customerId);
+        $customer = $this->getCustomerModel()->load($customerId);
         $storeId = $customer->getStoreId();
+        $helper = $this->makeHelper();
 
-        //update mailchimp ecommerce data for that customer
-        Mage::getModel('mailchimp/api_customers')->update($customerId, $storeId);
-        //update subscriber data if a subscriber with the same email address exists
-        Mage::getModel('mailchimp/api_subscribers')->update($customer->getEmail(), $storeId);
+        if ($helper->isSubscriptionEnabled($storeId)) {
+            //update subscriber data if a subscriber with the same email address exists
+            $this->getApiSubscriber()->update($customer->getEmail(), $storeId);
+        }
+
+        if ($helper->isEcomSyncDataEnabled($storeId)) {
+            //update mailchimp ecommerce data for that customer
+            $this->getApiCustomer()->update($customerId, $storeId);
+        }
+
+        return $observer;
     }
 
     /**
-     * When Product object is saved set it to be updated on MailChimp if getMailchimpUpdateObserverRan() is false.
+     * Set the products included the order to be updated on MailChimp on the next cron job run.
      *
      * @param  Varien_Event_Observer $observer
      * @return Varien_Event_Observer
+     * @throws Exception
      */
-    public function productSaveBefore(Varien_Event_Observer $observer)
+    public function newOrder(Varien_Event_Observer $observer)
     {
-        $product = $observer->getEvent()->getProduct();
-        $storeId = Mage::app()->getStore()->getStoreId();
-        //update mailchimp ecommerce data for that product variant
-        $this->makeApiProducts()->update($product->getId(), $storeId);
+        $helper = $this->makeHelper();
+        $post = $helper->getMageApp()->getRequest()->getPost('mailchimp_subscribe');
+        $order = $observer->getEvent()->getOrder();
+        $storeId = $order->getStoreId();
+        $ecommEnabled = $helper->isEcomSyncDataEnabled($storeId);
+
+        if ($ecommEnabled) {
+            if (isset($post)) {
+                $email = $order->getCustomerEmail();
+                $subscriber = $helper->loadListSubscriber($post, $email);
+                if ($subscriber) {
+                    $helper->subscribeMember($subscriber, true);
+                }
+            }
+
+            $this->removeCampaignData();
+
+            $items = $order->getAllItems();
+            foreach ($items as $item) {
+                if ($item->getProductType() == 'bundle' || $item->getProductType() == 'configurable') {
```

Comment:
use `Mage_Catalog_Model_Product_Type::TYPE_BUNDLE` and `Mage_Catalog_Model_Product_Type::TYPE_CONFIGURABLE`